src/bidiMapper/domains/context/browsingContextProcessor.ts

```diff
@@ -38,38 +38,63 @@ export class BrowsingContextProcessor {
     cdpConnection: CdpConnection,
     selfTargetId: string,
     onContextCreated: (t: Context) => Promise<void>,
-    onContextDestroyed: (t: Context) => Promise<void>
+    onContextDestroyed: (t: Context) => Promise<void>,
+    private EVALUATOR_SCRIPT: string
   ) {
     this._cdpConnection = cdpConnection;
     this._selfTargetId = selfTargetId;
     this._onContextCreated = onContextCreated;
     this._onContextDestroyed = onContextDestroyed;
+
+    this._setCdpEventListeners(this._cdpConnection.browserClient());
   }
 
+  private _setCdpEventListeners(browserCdpClient: CdpClient) {
+    browserCdpClient.Target.on('attachedToTarget', async (params) => {
+      await this._handleAttachedToTargetEvent(params);
+    });
+    browserCdpClient.Target.on('targetInfoChanged', (params) => {
+      this._handleInfoChangedEvent(params);
+    });
+    browserCdpClient.Target.on('detachedFromTarget', (params) => {
+      this._handleDetachedFromTargetEvent(params);
+    });
+  }
+
+  // Creation of `Context` can take quite a while. To avoid race condition, in
+  // the map kept Pmomise, eventualy resolved with the`Context`.
```

Comment:
```suggestion
  // Creation of `Context` can take quite a while. To avoid race condition, keep
  // a Promise in the map, eventually resolved with the `Context`.
```